draft-ietf-tls-dtls-connection-id.md

```diff
@@ -486,7 +497,8 @@ that is malicious. This attack is of concern when there is a large asymmetry
 of request/response message sizes. 
 
 Additionally, an attacker able to observe the data traffic exchanged between 
-two DTLS peers is able to replay datagrams with modified IP address/port numbers. 
+two DTLS peers is able to replay datagrams with modified IP address/port numbers
+if the optional DTLS replay detection functionality is not in use. 
```

Comment:
I think it would be better to just note that you can replace those packets. the replay point seems like a distraction.

Reply:
The replays will be accepted as valid, though, whereas arbitrary replacement would not.
Or are you saying that those *fields* can be replaced, not the packets themselves...

Reply:
what I'm saying is that you could have suppressed the packets and played them with your own IP and that attack would be just as good and isn't prevented by the DTLS anti-replay mechanism.

Reply:
> played them with your own IP

That "own IP" isn't used for too much, potentially to send a message back.
With "IP changes only on newer messages (according the record sequence number)", that narrows the attack to even that "newest package once", until an other message gets the newest (including the spoofed message itself). 
I tried to express my assumption about that attack in my comment to PR #71 .
For me it seems to be not to likely, that someone tries to DDoS using something, waiting passive for passing messages instead of active generating messages. That's mainly a question of the assumed message volume.

Reply:
I think the goal here is to describe what can be done; there may not need to be a great motivation for or benefit from doing so.

I took a harder think about about this and pushed some new text.  I think that it is still worth talking about the replay case, since the replays can be modified as well, but the straight-up modification case is more promiinent.

Reply:
Let me fist mention, that both, the new text and ekr's comment are pretty right.

But since more than a year it seems to be still unclear, if, beside an "amplification attack", that causes any more trouble? So, yes, that is a possible scenario, that someone is changing the ip-address. But with that, what will be the negative effect of that? The record will be processed and only the usage of that spoofed address to send something back seems to be the critical part.

FMPOV, it' s very easy:
if the ip-address is moved out of cryptographic validation with the CID,
-  then this is either very general a "no go", because it's not possible to limit the damage of it
- or it is possible to define the limits of that attacks.

So, is this going to define the limits? Or should we consider, that, if the ip-address is outside of the cryptographic validation, it's not possible to define that?

Reply:
> what I'm saying is that you could have suppressed the packets and played them with your own IP and that attack would be just as good and isn't prevented by the DTLS anti-replay mechanism.

Is the intention of this clarification, that the simple-cases, where either a "replay-protection" or a "newer" may help, should be removed and the "address validation" should be accented?

Reply:
My main goal here is to accentuate aspects that are different and possibly "surprising" than regular DTLS due to the addition of the CID (and no longer relying on 5-tuple).  In regular DTLS the attacker can't modify the IP addresses because they're a lookup key, so the ability to do that at all is noteworthy.  Likewise, for regular DTLS (without replay protection) an attacker can replay many copies, but the addresses in them (and thus the entire packet itself) have to be the same every time.  With CIDs, the replayed copies don't have to be bit-for-bit identical, which also seems noteworthy -- it's a "malleable" replay.  The specific consequences of this will vary from case to case, and the one "amplification" attack that we've identified so far is particularly frightening (and we have some partial mitigations for), but we should still state the new properties since we are not considering all potential use cases.  In other words, the goal here is not to specify exactly all sorts of "trouble" this can cause, but rather to note the difference from the previous state of affairs, so that the reader can think about how (if at all) it affects their use case.  With that mindset, I do not find the question of whether it is possible to define the "limits of that attacks" to be worth pursuing very much.

Reply:
> With CIDs, the replayed copies don't have to be bit-for-bit identical, which also seems noteworthy

Example?

Reply:
> In other words, the goal here is not to specify exactly all sorts of "trouble" this can cause, but rather to note the difference from the previous state of affairs, so that the reader can think about how (if at all) it affects their use case. 

> With that mindset, I do not find the question of whether it is possible to define the "limits of that attacks" to be worth pursuing very much.

I would prefer, if the first is discussed in the community, so that everybody can benefits from the accumulated result. That requires on my view, to address or discuss the considered "limits of that attacks".